src/Controls/src/Core/VisualStateManager.cs

```diff
@@ -378,7 +379,7 @@ public VisualStateGroup()
 		/// <include file="../../docs/Microsoft.Maui.Controls/VisualStateGroup.xml" path="//Member[@MemberName='CurrentState']/Docs/*" />
 		public VisualState CurrentState { get; internal set; }
 
-		internal VisualElement VisualElement { get; set; }
+		internal WeakReference<VisualElement> VisualElement { get; set; }
```

Comment:
Just an idea to make the code cleaner, you could keep the property unchanged and make the backing field a `WeakReference<VisualElement>`. Then the property getter could do null check and `TryGetValue()`.